### PR TITLE
Review

### PR DESCRIPTION
**Pull request by jaenek**
Zajęło ze 3 godzinki, nie licząc ustawienia środowiska.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,17 +14,45 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <maven.compiler.source>1.7</maven.compiler.source>
```
Comment (reevee-codes) [inline pom.xml:17]: how do you want to compile java 11 with java 1.7?

File: src/test/java/com/scs_test/AppTest.java [hunk 2]
```diff
@@ -1,20 +1,34 @@
...
+import io.restassured.module.jsv.JsonSchemaValidator;
 
 /**
- * Unit test for simple App.
+ * Unit test for Spring Currency Sample App.
  */
-public class AppTest 
-{
-    /**
-     * Rigorous Test :-)
```
Comment (reevee-codes) [inline src/test/java/com/scs_test/AppTest.java:13]: redundant

File: src/test/java/com/scs_test/AppTest.java [hunk 3]
```diff
@@ -1,20 +1,34 @@
...
-     */
+public class AppTest {
```
Comment (reevee-codes) [inline src/test/java/com/scs_test/AppTest.java:13]: class name

File: src/test/java/com/scs_test/AppTest.java [hunk 4]
```diff
@@ -1,20 +1,34 @@
...
+    public void testBasic(String endpoint, String schemaPath) {
+        RestAssured.given().filter(new RestAssuredRequestFilter()).when().get(endpoint).then().assertThat().body(JsonSchemaValidator.matchesJsonSchemaInClasspath(schemaPath)).extract().response();
+    }
+
+    @Test
+    public void testEndpoints() {
+        HashMap<String, String> endpoints = new HashMap<String,String>();
+        endpoints.put("/gold", "gold.json");
```
Comment (reevee-codes) [inline src/test/java/com/scs_test/AppTest.java:21]: lack of test cases, edge cases

File: src/test/java/com/scs_test/RestAssuredRequestFilter.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import io.restassured.specification.FilterableResponseSpecification;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class RestAssuredRequestFilter implements Filter {
+    private static final Logger logger = LogManager.getLogger(RestAssuredRequestFilter.class);
+
+    @Override
+    public Response filter(FilterableRequestSpecification requestSpec, FilterableResponseSpecification responseSpec, FilterContext ctx) {
+        Response response = ctx.next(requestSpec, responseSpec);
```
Comment (reevee-codes) [inline src/test/java/com/scs_test/RestAssuredRequestFilter.java:16]: is it possible to do it in easier way? still great and good job

File: .classpath [hunk 6]
```diff
@@ -13,7 +13,7 @@
 			<attribute name="test" value="true"/>
 		</attributes>
 	</classpathentry>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7">
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.6">
 		<attributes>
```
Comment (reevee-codes) [inline .classpath:17]: co cie sklonilo do wyboru eclipse zamiast intellij?

File: .mvn/wrapper/maven-wrapper.properties [hunk 7]
```diff
@@ -0,0 +1,2 @@
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.6.3/apache-maven-3.6.3-bin.zip
```
Comment (reevee-codes) [inline .mvn/wrapper/maven-wrapper.properties:1]: this is redundant

File: mvnw [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+esac
+
+if [ -z "$JAVA_HOME" ] ; then
+  if [ -r /etc/gentoo-release ] ; then
+    JAVA_HOME=`java-config --jre-home`
+  fi
+fi
+
+if [ -z "$M2_HOME" ] ; then
+  ## resolve links - $0 may be a link to maven's home
```
Comment (reevee-codes) [inline mvnw:76]: we can assume that we wont need to use wrapper

File: mvnw.cmd [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+    )
+    if "%MVNW_VERBOSE%" == "true" (
+        echo Couldn't find %WRAPPER_JAR%, downloading it ...
+        echo Downloading from: %DOWNLOAD_URL%
+    )
+
+    powershell -Command "&{"^
+		"$webclient = new-object System.Net.WebClient;"^
+		"if (-not ([string]::IsNullOrEmpty('%MVNW_USERNAME%') -and [string]::IsNullOrEmpty('%MVNW_PASSWORD%'))) {"^
+		"$webclient.Credentials = new-object System.Net.NetworkCredential('%MVNW_USERNAME%', '%MVNW_PASSWORD%');"^
```
Comment (reevee-codes) [inline mvnw.cmd:147]: a) this file shouldnt be here b) its not necessary

File: pom.xml [hunk 10]
```diff
@@ -14,17 +14,45 @@
...
+
+    <!-- https://mvnrepository.com/artifact/io.rest-assured/rest-assured -->
+    <dependency>
+      <groupId>io.rest-assured</groupId>
+      <artifactId>rest-assured</artifactId>
+      <version>4.4.0</version>
+      <scope>test</scope>
+    </dependency>
+
+    <!-- https://mvnrepository.com/artifact/io.rest-assured/json-schema-validator -->
```
Comment (reevee-codes) [inline pom.xml:37]: do not leave comments in pom but it also depends on the project

File: pom.xml [hunk 11]
```diff
@@ -14,17 +14,45 @@
...
 
   <dependencies>
+
     <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
       <version>4.11</version>
       <scope>test</scope>
     </dependency>
+
```
Comment (reevee-codes) [inline pom.xml:28]: sometimes its better to not leave empty spaces

File: .settings/org.eclipse.m2e.core.prefs [hunk 12]
```diff
@@ -0,0 +1,4 @@
+activeProfiles=
+eclipse.preferences.version=1
```
Comment (reevee-codes) [inline .settings/org.eclipse.m2e.core.prefs:2]: it shouldnt be downloaded and seen by someone who uses different IDE than eclipse

File: .classpath (hunk 6 above)
Comment (reevee-codes) [reply .classpath:17]: ofc nie mowie ze to zle, zastanawiam sie tylko gdzie sie uzywa eclipse, w C++? niestety w javie industry standard to bedzie intellij

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+                    }
+                } catch (IOException e) {
+                    // Ignore ...
+                }
+            }
+        }
+        System.out.println("- Downloading from: " + url);
+
+        File outputFile = new File(baseDirectory.getAbsolutePath(), MAVEN_WRAPPER_JAR_PATH);
+        if(!outputFile.getParentFile().exists()) {
```
Comment (reevee-codes) [inline .mvn/wrapper/MavenWrapperDownloader.java:79]: redundant file

File: src/test/java/com/scs_test/AppTest.java [hunk 14]
```diff
@@ -1,20 +1,34 @@
...
+            testBasic(endpoint, endpoints.get(endpoint));
+        }
+    }
+
     @Test
-    public void shouldAnswerWithTrue()
-    {
-        assertTrue( true );
+    public void testMid() {
+        RestAssured.given().filter(new RestAssuredRequestFilter()).when().queryParam("currency", "usd").get("/mid").then()
```
Comment (reevee-codes) [inline src/test/java/com/scs_test/AppTest.java:31]: great:)

File: src/test/resources/log4j2.properties [hunk 15]
```diff
@@ -0,0 +1,14 @@
+status = error
+name = Log4j2PropertiesConfig
+ 
+appenders = file
+ 
+appender.file.type = File
+appender.file.name = FileLogger
+appender.file.filename = logs.txt
```
Comment (reevee-codes) [inline src/test/resources/log4j2.properties:8]: what happens if we delete this file?

File: .classpath [hunk 16]
```diff
@@ -13,7 +13,7 @@
 			<attribute name="test" value="true"/>
 		</attributes>
 	</classpathentry>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7">
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.6">
```
Comment (reevee-codes) [inline .classpath:16]: why it got changed?

File: pom.xml [hunk 17]
```diff
@@ -14,17 +14,45 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <maven.compiler.source>1.7</maven.compiler.source>
-    <maven.compiler.target>1.7</maven.compiler.target>
```
Comment (reevee-codes) [inline pom.xml:18]: does it compile without this line?

File: .classpath [hunk 18]
```diff
@@ -13,7 +13,7 @@
 			<attribute name="test" value="true"/>
 		</attributes>
 	</classpathentry>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7">
```
Comment (reevee-codes) [inline .classpath:16]: should be added to gitignore as it is redundant for intellij users
Why gitignore is added only on master?
Reviewer will checkout your PR and might be angry and pulling random stuff xD
